File: ticketflow/gateway/src/logger.js
// Logger JSON minimale condiviso dal gateway.
const LEVELS = { error: 0, warn: 1, info: 2, debug: 3 };
const current = LEVELS[process.env.LOG_LEVEL] ?? LEVELS.info;
function log(level, message, meta = {}) {
  if (LEVELS[level] > current) return;
  process.stdout.write(JSON.stringify({ ts: new Date().toISOString(), level, service: 'gateway', message, ...meta }) + '\n');
}
export const logger = {
  error: (m, meta) => log('error', m, meta),
  warn: (m, meta) => log('warn', m, meta),
  info: (m, meta) => log('info', m, meta),
  debug: (m, meta) => log('debug', m, meta),
};

File: ticketflow/service-orders/app/logger.py
"""Logging JSON minimale, condiviso dai servizi FastAPI."""
import json
import logging
import os
import sys


class JsonFormatter(logging.Formatter):
    def __init__(self, service: str) -> None:
        super().__init__()
        self.service = service

    def format(self, record: logging.LogRecord) -> str:
        payload = {
            "ts": self.formatTime(record),
            "level": record.levelname.lower(),
            "service": self.service,
            "message": record.getMessage(),
        }
        return json.dumps(payload)


def get_logger(service: str) -> logging.Logger:
    logger = logging.getLogger(service)
    if not logger.handlers:
        handler = logging.StreamHandler(sys.stdout)
        handler.setFormatter(JsonFormatter(service))
        logger.addHandler(handler)
    logger.setLevel(os.getenv("LOG_LEVEL", "info").upper())
    return logger
